packages/kolibri-tools/lib/webpack_json.py

```diff
@@ -90,53 +53,60 @@ def expand_glob(build_item):
         [item for item in build_item.split(".") if item and item != "*"]
     )
     try:
-        parent_module = importlib.import_module(parent_module_path)
-    except ImportError:
-        # Someone has passed us an invalid module path, carry on
-        return plugins
-    for file in glob.glob(os.path.join(os.path.dirname(parent_module.__file__), "*")):
-        try:
-            file = file.replace(os.path.dirname(parent_module.__file__), "")
-            child_module_path = parent_module_path + file.replace("/", ".")
-            plugins.append(initialize_plugin(child_module_path))
-        except ImportError:
-            continue
+        for file in resource_listdir(parent_module_path, "."):
+            if resource_isdir(parent_module_path, file):
+                try:
+                    child_module_path = parent_module_path + "." + file
+                    plugins.append(plugin_data(child_module_path))
+                except ImportError:
+                    continue
+    except OSError:
+        pass
     return plugins
 
 
-def initialize_plugin(app, path=None):
-    try:
-        import_string = app + ".kolibri_plugin"
-        if path:
-            # Need to define the base module by its path first
-            import_package(app, os.path.join(path, "__init__.py"))
-            # Then import the plugin module after
-            plugin_module = import_package(
-                import_string, os.path.join(path, "kolibri_plugin.py")
+def plugin_data(module_path):
+    if resource_exists(module_path, "webpack.config.js"):
```

Comment:
this line is currently throwing an error for me when running `yarn run devserver`:

```
(kolibri) d@slate ~/P/l/kolibri (independent_build)> yarn run devserver
yarn run v1.15.2
$ npm-run-all --parallel django-devserver lint-frontend:watch:format hashi-dev watch services
npm WARN lifecycle The node binary used for scripts is /var/folders/9r/4w5cgtz95c5csjr86rv9qm_h0000gp/T/yarn--1565369704098-0.5163607372435111/node but npm is using /Users/d/.nvm/versions/node/v10.15.3/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
npm WARN lifecycle The node binary used for scripts is /var/folders/9r/4w5cgtz95c5csjr86rv9qm_h0000gp/T/yarn--1565369704098-0.5163607372435111/node but npm is using /Users/d/.nvm/versions/node/v10.15.3/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> kolibri-root@0.12.0-dev.1 django-devserver /Users/d/Projects/le/kolibri
> kolibri --debug manage runserver --settings=kolibri.deployment.default.settings.dev "0.0.0.0:8000" 


> kolibri-root@0.12.0-dev.1 lint-frontend:watch:format /Users/d/Projects/le/kolibri
> yarn run lint-frontend --monitor --write

npm WARN lifecycle The node binary used for scripts is /var/folders/9r/4w5cgtz95c5csjr86rv9qm_h0000gp/T/yarn--1565369704098-0.5163607372435111/node but npm is using /Users/d/.nvm/versions/node/v10.15.3/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> kolibri-root@0.12.0-dev.1 hashi-dev /Users/d/Projects/le/kolibri
> yarn workspace hashi run dev

npm WARN lifecycle The node binary used for scripts is /var/folders/9r/4w5cgtz95c5csjr86rv9qm_h0000gp/T/yarn--1565369704098-0.5163607372435111/node but npm is using /Users/d/.nvm/versions/node/v10.15.3/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> kolibri-root@0.12.0-dev.1 watch /Users/d/Projects/le/kolibri
> kolibri-tools build dev --file ./build_tools/build_plugins.txt

npm WARN lifecycle The node binary used for scripts is /var/folders/9r/4w5cgtz95c5csjr86rv9qm_h0000gp/T/yarn--1565369704098-0.5163607372435111/node but npm is using /Users/d/.nvm/versions/node/v10.15.3/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> kolibri-root@0.12.0-dev.1 services /Users/d/Projects/le/kolibri
> kolibri services --foreground --skipupdate

$ kolibri-tools lint '{kolibri*/**/assets,packages}/**/*.{js,vue,scss,less,css}' --ignore '**/node_modules/**','**/static/**','**/packages/kolibri-core-for-export/**' --monitor --write
WARNING:root:No C Extensions available for this platform.

WARNING:root:No C Extensions available for this platform.

$ yarn run build-base --mode=development --watch
$ webpack --config ./webpack.config.js --mode=development --watch
Kolibri Linter: Initializing watcher for the following patterns: {kolibri*/**/assets,packages}/**/*.{js,vue,scss,less,css}
INFO     Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE
INFO     Option RUN_MODE in section [Deployment] being overridden by environment variable KOLIBRI_RUN_MODE

webpack is watching the files…

WARNING:root:No C Extensions available for this platform.

Traceback (most recent call last):
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 177, in <module>
    main()
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 162, in main
    result = initialize_plugins(build_list)
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 106, in initialize_plugins
    plugins += expand_glob(build_item)
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 60, in expand_glob
    plugins.append(plugin_data(child_module_path))
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 69, in plugin_data
    if resource_exists(module_path, "webpack.config.js"):
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1186, in resource_exists
    return get_provider(package_or_requirement).has_resource(resource_name)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 406, in get_provider
    return _find_adapter(_provider_factories, loader)(module)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1439, in __init__
    self.module_path = os.path.dirname(getattr(module, "__file__", ""))
  File "/Users/d/Projects/le/kolibri/.venv/bin/../lib/python3.7/posixpath.py", line 156, in dirname
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType
child_process.js:656
    throw err;
    ^

Error: Command failed: PATH=$(echo $PATH | sed 's/\/usr\/bin://g')":/usr/bin" python /Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py --output_file /var/folders/9r/4w5cgtz95c5csjr86rv9qm_h0000gp/T/11979-79645-i0wsdn.2gc1.json --plugin_file ./build_tools/build_plugins.txt
WARNING:root:No C Extensions available for this platform.

Traceback (most recent call last):
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 177, in <module>
    main()
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 162, in main
    result = initialize_plugins(build_list)
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 106, in initialize_plugins
    plugins += expand_glob(build_item)
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 60, in expand_glob
    plugins.append(plugin_data(child_module_path))
  File "/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/webpack_json.py", line 69, in plugin_data
    if resource_exists(module_path, "webpack.config.js"):
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1186, in resource_exists
    return get_provider(package_or_requirement).has_resource(resource_name)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 406, in get_provider
    return _find_adapter(_provider_factories, loader)(module)
  File "/Users/d/Projects/le/kolibri/.venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 1439, in __init__
    self.module_path = os.path.dirname(getattr(module, "__file__", ""))
  File "/Users/d/Projects/le/kolibri/.venv/bin/../lib/python3.7/posixpath.py", line 156, in dirname
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType

    at checkExecSyncError (child_process.js:616:11)
    at execSync (child_process.js:653:13)
    at module.exports (/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/read_webpack_json.js:72:5)
    at Command.<anonymous> (/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/cli.js:132:24)
    at Command.listener (/Users/d/Projects/le/kolibri/node_modules/commander/index.js:315:8)
    at Command.emit (events.js:189:13)
    at Command.parseArgs (/Users/d/Projects/le/kolibri/node_modules/commander/index.js:651:12)
    at Command.parse (/Users/d/Projects/le/kolibri/node_modules/commander/index.js:474:21)
    at checkVersion (/Users/d/Projects/le/kolibri/packages/kolibri-tools/lib/cli.js:342:13)
    at /Users/d/Projects/le/kolibri/node_modules/check-node-version/index.js:133:7
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! kolibri-root@0.12.0-dev.1 watch: `kolibri-tools build dev --file ./build_tools/build_plugins.txt`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the kolibri-root@0.12.0-dev.1 watch script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/d/.npm/_logs/2019-08-09T16_55_08_135Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! kolibri-root@0.12.0-dev.1 hashi-dev: `yarn workspace hashi run dev`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the kolibri-root@0.12.0-dev.1 hashi-dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! kolibri-root@0.12.0-dev.1 lint-frontend:watch:format: `yarn run lint-frontend --monitor --write`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the kolibri-root@0.12.0-dev.1 lint-frontend:watch:format script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/d/.npm/_logs/2019-08-09T16_55_08_275Z-debug.log

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/d/.npm/_logs/2019-08-09T16_55_08_276Z-debug.log
ERROR: "watch" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
Hrm, wonder if reading the lines is happening differently on a Mac and it is getting a None somewhere.

Just need to add a `None` catch at the top of the function I think. Updating now.